### PR TITLE
Add Onnxruntime parity checks to CI

### PR DESCRIPTION
**Pull request by TedThemistokleous**
Add in onnxruntime parity checks into our CI after we build onnxruntime. Goal with this changeset is to ensure the tests run without error/issue and isn't concerned about performance. 

Parallelize the Parity tests, with the existing unit tests run and organize this so we can keep unit, install, and parity checks separate in our pipeline. ORT is now built with --build_wheel which is used with the final two stages (unit tests and parity)

Related to the issues seen in DLM where parity tests fail when we move ROCm versions. This change is to help get ahead of things

Related issues 

#1935  & #1877

### CODE REVIEW

File: tools/build_and_install_ort.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+cd /onnxruntime
+pip3 install -r requirements-dev.txt
+# Add newer cmake to the path
+export PATH="/opt/cmake/bin:$PATH"
+export CXXFLAGS="-D__HIP_PLATFORM_AMD__=1 -w"
+./build.sh --config Release  --cmake_extra_defines CMAKE_HIP_COMPILER=/opt/rocm/llvm/bin/clang++ --update --build --build_wheel --parallel --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) --skip_tests --rocm_home /opt/rocm --use_migraphx --migraphx_home /opt/rocm --rocm_version=`cat /opt/rocm/.info/version-dev` --allow_running_as_root
+
+#install onnxruntime wheel for parity tests
+cd build/Linux/Release
+pip3 install dist/*.whl
```
Comment (umangyadav) [inline tools/build_and_install_ort.sh:37]: add new line

File: Jenkinsfile [hunk 2]
```diff
@@ -164,13 +164,40 @@ def onnxnode(name, body) {
...
     }
 }
+
+rocmtest onnxtest: onnxtestnode('anygpu') {cmake_build ->
+    stage("Onnxruntime Unit tests") {
+        sh '''
+            cd /onnxruntime && ./test_onnxrt_unit_tests.sh 
+        '''
+    },
+    stage("Onnxruntime Unit tests") {
```
Comment (umangyadav) [inline Jenkinsfile:198]:
```suggestion
    stage("Onnxruntime Parity tests") {
```

Comment (TedThemistokleous) [reply Jenkinsfile]: Done

File: tools/build_and_install_ort.sh (hunk 1 above)
Comment (TedThemistokleous) [reply tools/build_and_install_ort.sh]: Done

File: tools/test_onnxrt_parity_checks.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#####################################################################################
+set -e
+
+#install dependancies used for parity checks
+pip3 install psutil onnx coloredlogs packaging transformers==4.29.2 sympy
+pip3 install -U numpy==1.21.6
```
Comment (pfultz2) [inline tools/test_onnxrt_parity_checks.sh:28]: These should be installed in the docker file. Some of these are pretty big dependencies we probably dont want to install everytime.

File: tools/test_onnxrt_parity_checks.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#####################################################################################
+set -e
+
+#install dependancies used for parity checks
+pip3 install psutil onnx coloredlogs packaging transformers==4.29.2 sympy
```
Comment (pfultz2) [inline tools/test_onnxrt_parity_checks.sh:27]: The versions should be pinned.

File: tools/test_onnxrt_parity_checks.sh (hunk 3 above)
Comment (TedThemistokleous) [reply tools/test_onnxrt_parity_checks.sh:28]: Sure I can move these forward

File: tools/test_onnxrt_parity_checks.sh (hunk 4 above)
Comment (TedThemistokleous) [reply tools/test_onnxrt_parity_checks.sh]: Done

File: tools/test_onnxrt_parity_checks.sh (hunk 3 above)
Comment (TedThemistokleous) [reply tools/test_onnxrt_parity_checks.sh]: Done

**Comment by TedThemistokleous** [comment]
odd seeing 
```

[2023-08-06T19:08:42.223Z] Caused by: hudson.plugins.git.GitException: Command "git init /home/jenkins/workspace/_AMDMIGraphX_add_parity_check_ci" returned status code 128:

[2023-08-06T19:08:42.223Z] stdout: 

[2023-08-06T19:08:42.223Z] stderr: error: copy-fd: write returned: No space left on device

[2023-08-06T19:08:42.223Z] fatal: cannot copy '/usr/share/git-core/templates/hooks/pre-rebase.sample' to '/data/workspace/_AMDMIGraphX_add_parity_check_ci/.git/hooks/pre-rebase.sample': No space left on device```

File: tools/test_onnxrt_parity_checks.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#####################################################################################
+set -e
+
+#Go to folder with parity checks
+cd /workspace/onnxruntime/onnxruntime/test/python/transformers/
+
+#Install latest stable torch version
+pip install torch torchvision torchaudio --index-url https://download.pytorch.org/whl/rocm5.4.2
```
Comment (causten) [inline tools/test_onnxrt_parity_checks.sh:30]: Is 5.4.2 really the latest stable? Would this even be compatible with ROCm 5.7 or 6.0?  How useful would this be if the version of migx being tested is 5.7 or 6.0?

File: tools/test_onnxrt_unit_tests.sh [hunk 6]
```diff
@@ -1,7 +1,7 @@
 #####################################################################################
 # The MIT License (MIT)
 #
-# Copyright (c) 2015-2022 Advanced Micro Devices, Inc. All rights reserved.
+# Copyright (c) 2015-2023 Advanced Micro Devices, Inc. All rights reserved.
```
Comment (causten) [inline tools/test_onnxrt_unit_tests.sh:4]: License checker will flag this.  Change to 2024

File: tools/test_onnxrt_parity_checks.sh (hunk 5 above)
Comment (TedThemistokleous) [reply tools/test_onnxrt_parity_checks.sh:30]: I haven't touched this in a while I can update.

**Comment by TedThemistokleous** [comment]
Closing this out. Superceeded by more recent syncs to Onnxruntime main

**Pull request closed without merging**